pytorch_lightning/callbacks/base.py

```diff
@@ -33,6 +33,21 @@ class Callback(abc.ABC):
     Subclass this class and override any of the relevant hooks
     """
 
+    @property
+    def state_id(self) -> str:
+        """
+        Identifier for the state of the callback. Used to store and retrieve a callback's state from the
+        checkpoint dictionary by ``checkpoint["callbacks"][state_id]``. Implementations of a callback need to
+        provide a unique state id if 1) the callback has state and 2) it is desired to maintain the state of
+        multiple instances of that callback.
+        """
+        return self.__class__.__qualname__
```

Comment:
This still has the issue that if the class is moved to a different package, the qualname will not match, correct?

Reply:
if the class is renamed then yes, the qual name will change. if it get's moved it does not matter. 

It will depend who is the author of the callback. If it is us, the PL team, then we can write a migration within Lightning to provide BC, see #8558. If the author is the user, they will have to take care themselves.

Reply:
And this here is just the default implementation. See here for how certain callbacks could implement the state_id: https://github.com/PyTorchLightning/pytorch-lightning/pull/7187/files